src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,129 @@
+import clsx from "clsx";
+import { Button } from "components/Button";
+import { Menu, MenuButton, MenuItem, type MenuProps } from "components/Menu";
+import type { MenuChildrenType } from "components/Menu/MenuTypes";
+import { type DetailedHTMLProps, type HTMLAttributes, useMemo } from "react";
+import type { IconNamesType } from "utils";
+
+import "./SplitButton_shim.css";
+
+/**
+ * SplitButtonProps
+ * @prop {MenuChildrenType} children - The menu items to display in the dropdown.
+ * @prop {string} [className] - The class name to add to the component.
+ * @prop {"sm" | "md"} [height] - The height of the SplitButton.
+ * @prop {MenuProps} [menuProps] - The props to pass to the Menu component.
+ * @prop {ButtonProps} buttonProps - The props to pass to the Button component.
+ */
+type AtLeastOneProps =
+	| { text: string; ariaLabel?: string; icon?: IconNamesType }
+	| { text?: string; ariaLabel: string; icon: IconNamesType };
+type SplitButtonButtonProps = {
+	variant?: "primary" | "secondary";
+	onClick?: () => void;
```

Comment:
I recommend making this required. I don't think it makes sense to have a button that doesn't have an `onClick` event. 
```suggestion
	onClick: () => void;
```

Reply:
agree.